tests/unit/backend/chalice/api_server/test_v1_collection.py

```diff
@@ -427,8 +430,143 @@ def test__list_collection__check_owner(self):
+        self.assertEqual(response.status_code, 202)
+
+        # check collection and datasets tombstoned
+        self.session.expire_all()
+        collection = Collection.get_collection(
+            self.session, collection.id, CollectionVisibility.PRIVATE.name, include_tombstones=True
+        )
+
+        self.assertTrue(collection.tombstone)
+        self.assertTrue(dataset_1.tombstone)
+        self.assertTrue(dataset_2.tombstone)
+
+        response = self.app.get(test_url.url, headers=headers)
+        self.assertEqual(response.status_code, 403)
+
+    def test_delete_collection__dataset_not_available(self):
+        # Generate test collection
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="test_user_id"
+        )
+        processing_status = {"upload_status": UploadStatus.UPLOADED, "upload_progress": 100.0}
+
+        dataset = self.generate_dataset(self.session, collection=collection, processing_status=processing_status)
         headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
-        response = self.app.delete(path.url, headers)
+        dataset_url = furl(path=f"/dp/v1/datasets/{dataset.id}/status")
+        response = self.app.get(dataset_url.url, headers=headers)
+        self.assertEqual(response.status_code, 200)
+
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        response = self.app.delete(test_url.url, headers=headers)
         response.raise_for_status()
+        self.assertEqual(response.status_code, 202)
+
+        response = self.app.get(dataset_url.url, headers=headers)
+        self.assertEqual(response.status_code, 403)
+
+    def test_delete_collection__already_tombstoned__ok(self):
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="test_user_id", tombstone=True
+        )
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url.url, headers=headers)
+        self.assertEqual(response.status_code, 202)
+
+    def test_delete_collection__public(self):
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PUBLIC.name, owner="test_user_id"
+        )
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        response = self.app.delete(test_url.url, headers=headers)
+        self.assertEqual(response.status_code, 403)
```

Comment:
```suggestion
        self.assertEqual(response.status_code, 405)
```

Reply:
based on the changes you recommend above (removing visibility as a param) this will return a 403 not a separate error